docs/RecoveringFromDeepForks.md

```diff
@@ -0,0 +1,67 @@
+# Recovering from deep forks
+
+Help! My node has failed with the following fatal error:
+
+```
+Insert error message here
+```
+
+This means that a peer sent our node a new cut, weightier than our current one,
+that has a greatest common ancestor of our current cut that is more than
+`reorg-limit` blocks behind:
+
+
+```
+                     (htA, hashA)
+   ... history ... - o - o - o - ... - o
+                     |                 (htOld, hashOld)
+                     |
+                     ` - o - o - ... - o
+                                       (htNew, hashNew)
+```
+
+Here the node will fail with a fatal error if `(htOld - htA) >= reorg-limit`,
+because huge forks like this are very unlikely absent some software problem or
+attack situation, both requiring operator intervention.
+
+## Recovery method A: get DB files from a friendly peer
+
+If you can get up-to-date database files for the blockchain from a friendly
+peer or from Kadena support, this may be the fastest way to catch up a lagging
+or forked node.
+
+```
+home# ssh root@remote-node
+remote-node# systemctl stop chainweb
+remote-node# tar cvzf chainweb-db.tar.gz .local/share/chainweb-node
+remote-node# systemctl start chainweb
+remote-node# exit
+home# systemctl stop chainweb
+home# rm -Rf ~/.local/share/chainweb-node
+home# ssh root@remote-node 'cat chainweb.db.tar.gz' | tar xvzf -
+home# systemctl start chainweb
+```
+
+## Recovery method B: restart node with a cut height limit
+
+This option is slower, but probably safer/easier. The error message will report
+an ancestor blockheight on a particular chain; your goal is to select a
+cumulative cut height that will rewind us behind this block. Select:
+
+```
+cut_height_limit = (height_ancestor - 50) * num_chains
+```
+
+And restart the node with the following changes to `chainweb.yaml`:
+
+```
+chainweb:
+  cuts:
+    pruneChainDatabase: false
+    initialCutHeightLimit: $cut_height_limit
+```
```

Comment:
one should use the `private` flag and only provide know-peers that are on the "good" target fork.

Reply:
This `private` flag is important in deep fork scenarios with "weight inversion". A weight inversion is the situation where a fork exists that has been heavier in the past than the currently heaviest branch.

Normally, this scenario should be even much more rare than deep forks. While deep forks can happen as a consequence of long-lasting network partitions, deep forks with weight inversion can happen as a consequence of long lasting multiple overlapping partitions of the network that got resolved at different times. Forks with weight inversion can also happen on network partitions where the hash power within a partition changes a lot.

In the past forks with weight inversions could also be a consequence of emergency DAs, but that option doesn't exist any more.